new github-centric release process attempt

orndorffgrant opened the pull request:
This attempts to document a new github-centric release process. This is mostly modeled on what we did for version 31, but some of it isn't totally tested yet. Please let me know what you think.

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -10,12 +10,22 @@ The release process for ubuntu-advantage-tools has three overarching steps/goals
-An average release should take somewhere between 10 and 14 calendar days if things go smoothly, starting at the decision to release and ending at the new version being available in all supported Ubuntu releases. Note that it is not 2 weeks of full time work. Most of the time is spent waiting for review or sitting in proposed.
+Releases are scheduled to take 6 weeks, following the schedule below:
+- Week 1
+  - pro-client maintainer sets up the new version
```
```suggestion
  - A Pro Client maintainer sets up the new version
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -10,12 +10,22 @@ The release process for ubuntu-advantage-tools has three overarching steps/goals
+- Week 1
+  - pro-client maintainer sets up the new version
+  - Sponsor from Canonical Server team reviews
+- Weeks 2 and 3: SRU review ending in upload to devel and -proposed at the same time
```
```suggestion
- Weeks 2 and 3: SRU review ending in upload to `-devel` and `-proposed` at the same time
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -10,12 +10,22 @@ The release process for ubuntu-advantage-tools has three overarching steps/goals
+  - pro-client maintainer sets up the new version
+  - Sponsor from Canonical Server team reviews
+- Weeks 2 and 3: SRU review ending in upload to devel and -proposed at the same time
+- Week 4: Sit in -proposed for 7 days. Verification happens during this time.
```
```suggestion
- Week 4: Sit in `-proposed` for 7 days. Verification happens during this time.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -10,12 +10,22 @@ The release process for ubuntu-advantage-tools has three overarching steps/goals
+- Weeks 2 and 3: SRU review ending in upload to devel and -proposed at the same time
+- Week 4: Sit in -proposed for 7 days. Verification happens during this time.
+- Weeks 5 and 6: These weeks are allocated as a buffer in case any of the previous steps take longer than expected
+- After week 6: Release to -updates
```
```suggestion
- After week 6: Release to `-updates`
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
-
-> **Note**
-> `kinetic` is used throughout as a reference to the current devel release. This will change.
+### Git branch release strategy overview
+This diagram presents the branches that happen during the process of releasing version 42.
```
```suggestion
### Git branch release strategy overview

This diagram presents the branches that happen during the process of releasing version 42.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+When we reach a point in `main` where we are ready to release version 42, we do the following:
+* Tag that commit as `42-rc`
+* Create a branch called `review-v42` from the `41` tag
+  * This branch does not have any new commits on it, it is only used a as a target for the PR that will be created in the next step
```
```suggestion
  * This branch does not have any new commits on it, it is only used as a target for the PR that will be created in the next step
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+* Sponsor and SRU review uses the `main`->`review-v42` PR
+  * Feedback that requires changes are added to `main`
+* When version 42 is approved, the tip of `main` is tagged `42` and the Sponsor uploads using that commit
+  * The `review-v42` branch is now deleted along with the `main`->`review-v42` PR (The PR is not merged).
```
```suggestion
  * The `review-v42` branch is now deleted along with the `main`->`review-v42` PR (the PR is not merged).
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+
+You don't need to include all of the commits generated. Remember that the changelog should be read by the user to understand the new features/modifications in the package. If you think a commit will not add that much to the user experience, you can drop it from the changelog.
+
+To structure the changelog you can use the other entries as example. But we basically try to keep this order: debian changes, new features/modifications, testing. Within each section, bullet points should be alphabetized.
```
```suggestion
To structure the changelog you can use the other entries as example. But we basically try to keep this order: Debian changes; new features/modifications; testing. Within each section, bullet points should be alphabetised.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+  * Is this entry redundant? Sometimes we may have changes that affect separate modules of the
+    code. We should have an entry only for the module that was most affected by it.
+  * Is the changelog entry unique? We need to verify that the changelog entry is not already
+    reflected in an earlier version of the changelog. If it is, we need not only to remove but double
+    check the process we are using to cherry-pick the commits.
```
```suggestion
    reflected in an earlier version of the changelog. If it is, we need to not only remove but also to double
    check the process we are using to cherry-pick the commits.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+### 3. Release to our staging PPA
+
+1. Clone the repository into a clean directory and `git checkout $version`.
+    * *WARNING* Build the package in a clean environment. The reason is that the package
+    will contain everything that is present in the folder. If you are storing credentials or
```
```suggestion
    * *WARNING* Build the package in a clean environment, because the package
    will contain everything that is present in the folder. If you are storing credentials or
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+    * *WARNING* Build the package in a clean environment. The reason is that the package
+    will contain everything that is present in the folder. If you are storing credentials or
+    other sensible development information in your folder, they will be uploaded too when we send
+    the package to the ppa. A clean environment is the safest way to perform this.
```
```suggestion
    the package to the PPA. A clean environment is the safest way to perform this.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+4. Run `sbuild-it ../out/<package_name>.dsc`
+  * If this succeeds move on. If this fails, debug and fix before continuing.
+
+5. Repeat steps 2 through 4 for all supported Ubuntu Releases
```
```suggestion
5. Repeat steps 2 through 4 for all supported Ubuntu releases
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+5. Repeat steps 2 through 4 for all supported Ubuntu Releases
+  * The version for series other than devel should be in the form `<version>~<ubuntu-release-number>~rc<release-candidate-number>`
+  This means you must add the release number in the changelog. For example, suppose
+  the devel version is `1.1~rc1`. If you want to build for jammy now, change it to
```
```suggestion
  the devel version is `1.1~rc1`. If you want to build for Jammy now, change it to
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+  the devel version is `1.1~rc1`. If you want to build for jammy now, change it to
+  `1.1~22.04~rc1`.
+
+6. For each release, dput to the staging PPA:
```
```suggestion
6. For each release, `dput` to the staging PPA:
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+### 4. Release to Ubuntu (devel and SRU)
 
 1. Prepare SRU Launchpad bugs.
+    * We do this even before a successful merge into ubuntu/devel because the context added to these bugs is useful for the Server Team reviewer.
```
```suggestion
    * We do this even before a successful merge into `ubuntu/devel` because the context added to these bugs is useful for the Server Team reviewer.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+  * If this succeeds move on. If this fails, debug and fix before continuing.
+
+5. Repeat steps 2 through 4 for all supported Ubuntu Releases
+  * The version for series other than devel should be in the form `<version>~<ubuntu-release-number>~rc<release-candidate-number>`
```
```suggestion
  * The version for series other than `devel` should be in the form `<version>~<ubuntu-release-number>~rc<release-candidate-number>`
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+2. Edit the changelog
+  * List yourself as the author of this release.
+  * Edit the version number to look like: `$version~rc1` (`<version>~rc<release-candidate-number>`)
+  * Edit the Ubuntu release name. Start with the ubuntu/devel release.
```
```suggestion
  * Edit the Ubuntu release name. Start with the `ubuntu/devel` release.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
-      * e.g. `git rebase --onto pkg/ubuntu/devel 27.0.2 27.1`
-      * You may need to resolve conflicts, but hopefully these will be minimal.
-      * You'll end up in a detached state.
+      * For the testing steps, include steps to reproduce the bug. Then include instructions for adding `ppa:ua-client/staging`, and steps to verify the bug is no longer present. Mention that the staging PPA will be replaced with -proposed for actual verification.
```
```suggestion
      * For the testing steps, include steps to reproduce the bug. Then include instructions for adding `ppa:ua-client/staging`, and steps to verify the bug is no longer present. Mention that the staging PPA will be replaced with `-proposed` for actual verification.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
````diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
-    h. With the packages published to `ppa:ua-client/staging`, add links to the autopkgtest triggers to the Merge Proposal. The reviewer will have permission to trigger those tests. The links can be obtained by running `ppa tests -r <release> -a <arch1,arch2> ua-client/staging -L`
+    * If there is anything particularly interesting about this PR that a reviewer may want to know, put it here.
+    ```
+    * Add links to the autopkgtest triggers for the staging PPA to the PR. The reviewer will have permission to trigger those tests. The links can be obtained by running `ppa tests -r <release> -a <arch1,arch2> ua-client/staging -L`
````
```suggestion
    * Add links to the `autopkgtest` triggers for the staging PPA to the PR. The reviewer will have permission to trigger those tests. The links can be obtained by running `ppa tests -r <release> -a <arch1,arch2> ua-client/staging -L`
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
-      * After the release is finished, any commits that were merged directly into the release branch in this way should be brought back into `main` via a single PR.
-
-    c. Once review is complete and approved, the Server Team member should **not** upload the version to the devel release.
+    * Once Sponsor review is complete and approved, the Sponsor should **not** upload the version to the devel release.
```
```suggestion
    * Once Sponsor review is complete and approved, the Sponsor should **not** upload the version to the `devel` release.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+    * When the Sponsor is happy, tag the SRU reviewer again and let them know it is their turn to review. Follow the same process for addressing feedback from the SRU reviewer.
+
+
+4. Approval and upload to devel and -proposed unapproved queue
```
```suggestion
4. Approval and upload to `devel` and `-proposed` unapproved queue
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
````diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+      -- Grant Orndorff <grant.orndorff@canonical.com>  Thu, 29 Feb 2024 09:03:11 -0500
+      ```
+      * The versions for the stable releases must be in the format `$version~<release-number>`
+    * Tell the Sponsor that the branches are ready for them to upload to devel and the SRU unapproved queue.
````
```suggestion
    * Tell the Sponsor that the branches are ready for them to upload to `devel` and the SRU unapproved queue.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
````diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+      ```
+      * The versions for the stable releases must be in the format `$version~<release-number>`
+    * Tell the Sponsor that the branches are ready for them to upload to devel and the SRU unapproved queue.
+    * Check the [-proposed release queue](https://launchpad.net/ubuntu/xenial/+queue?queue_state=1&queue_text=ubuntu-advantage-tools) for presence of ubuntu-advantage-tools in unapproved state for each supported release. Note: libera chat #ubuntu-release IRC channel has a bot that reports queued uploads of any package in a message like "Unapproved: ubuntu-advantage-tools .. version".
````
```suggestion
    * Check the [`-proposed` release queue](https://launchpad.net/ubuntu/xenial/+queue?queue_state=1&queue_text=ubuntu-advantage-tools) for presence of `ubuntu-advantage-tools` in unapproved state for each supported release. Note: Libera chat `#ubuntu-release` IRC channel has a bot that reports queued uploads of any package in a message like `"Unapproved: ubuntu-advantage-tools .. version"`.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+      * The versions for the stable releases must be in the format `$version~<release-number>`
+    * Tell the Sponsor that the branches are ready for them to upload to devel and the SRU unapproved queue.
+    * Check the [-proposed release queue](https://launchpad.net/ubuntu/xenial/+queue?queue_state=1&queue_text=ubuntu-advantage-tools) for presence of ubuntu-advantage-tools in unapproved state for each supported release. Note: libera chat #ubuntu-release IRC channel has a bot that reports queued uploads of any package in a message like "Unapproved: ubuntu-advantage-tools .. version".
+    * Tell the SRU Reviewer that the packages are in the -proposed unapproved queue. They will need to actually approve the package to move into -proposed.
```
```suggestion
    * Tell the SRU Reviewer that the packages are in the `-proposed` unapproved queue. They will need to actually approve the package to move into `-proposed`.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
 4. -proposed verification and release to -updates
 
-    a. As soon as the SRU vanguard approves the packages, a bot in #ubuntu-release will announce that ubuntu-advantage-tools is accepted into the applicable -proposed pockets, or the [Xenial -proposed release rejection queue](https://launchpad.net/ubuntu/xenial/+queue?queue_state=4&queue_text=ubuntu-advantage-tools) will contain a reason for rejections. Double check the SRU process bug for any ac ...
+    * As soon as the SRU vanguard approves the packages, a bot in #ubuntu-release will announce that ubuntu-advantage-tools is accepted into the applicable -proposed pockets, or the [Xenial -proposed release rejection queue](https://launchpad.net/ubuntu/xenial/+queue?queue_state=4&queue_text=ubuntu-advantage-tools) will contain a reason for rejections. Double check the SRU process bug for any act ...
```
```suggestion
    * As soon as the SRU vanguard approves the packages, a bot in `#ubuntu-release` will announce that `ubuntu-advantage-tools` is accepted into the applicable `-proposed` pockets, or the [Xenial `-proposed` release rejection queue](https://launchpad.net/ubuntu/xenial/+queue?queue_state=4&queue_text=ubuntu-advantage-tools) will contain a reason for rejection. Double check the SRU process bug for any actionable review feedback.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,215 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
 
-    a. As soon as the SRU vanguard approves the packages, a bot in #ubuntu-release will announce that ubuntu-advantage-tools is accepted into the applicable -proposed pockets, or the [Xenial -proposed release rejection queue](https://launchpad.net/ubuntu/xenial/+queue?queue_state=4&queue_text=ubuntu-advantage-tools) will contain a reason for rejections. Double check the SRU process bug for any ac ...
+    * As soon as the SRU vanguard approves the packages, a bot in #ubuntu-release will announce that ubuntu-advantage-tools is accepted into the applicable -proposed pockets, or the [Xenial -proposed release rejection queue](https://launchpad.net/ubuntu/xenial/+queue?queue_state=4&queue_text=ubuntu-advantage-tools) will contain a reason for rejections. Double check the SRU process bug for any act ...
       * Once accepted into `-proposed` by an SRU vanguard [ubuntu-advantage-tools shows up in the pending_sru page](https://people.canonical.com/~ubuntu-archive/pending-sru.html), check `rmadison ubuntu-advantage-tools | grep -proposed` to see if the upload exists in -proposed yet.
```
```suggestion
      * Once accepted into `-proposed` by an SRU vanguard [`ubuntu-advantage-tools` shows up in the `pending_sru` page](https://people.canonical.com/~ubuntu-archive/pending-sru.html), check `rmadison ubuntu-advantage-tools | grep -proposed` to see if the upload exists in `-proposed` yet.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -216,20 +267,34 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+    * Change the tags on the bug from `verification-needed` to `verification-done` (including the verification tags for each Ubuntu release).
 
-    e. For any other related Launchpad bugs that are fixed in this release, perform the verification steps necessary for those bugs and mark them `verification-done` as needed. This will likely involve following the test steps, but instead of adding the staging PPA, enabling -proposed.
+    * For any other related Launchpad bugs that are fixed in this release, perform the verification steps necessary for those bugs and mark them `verification-done` as needed. This will likely involve following the test steps, but instead of adding the staging PPA, enabling -proposed.
```
```suggestion
    * For any other related Launchpad bugs that are fixed in this release, perform the verification steps necessary for those bugs and mark them `verification-done` as needed. This will likely involve following the test steps, but instead of adding the staging PPA, enabling `-proposed`.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -216,20 +267,34 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+    * For any other related Launchpad bugs that are fixed in this release, perform the verification steps necessary for those bugs and mark them `verification-done` as needed. This will likely involve following the test steps, but instead of adding the staging PPA, enabling -proposed.
 
-    f. Once all SRU bugs are tagged as `verification*-done`, all SRU-bugs should be listed as green in [the pending_sru page](https://people.canonical.com/~ubuntu-archive/pending-sru.html).
+    * Once all SRU bugs are tagged as `verification*-done`, all SRU-bugs should be listed as green in [the pending_sru page](https://people.canonical.com/~ubuntu-archive/pending-sru.html).
```
```suggestion
    * Once all SRU bugs are tagged as `verification*-done`, all SRU-bugs should be listed as green in [the `pending_sru` page](https://people.canonical.com/~ubuntu-archive/pending-sru.html).
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -216,20 +267,34 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+    * Once all SRU bugs are tagged as `verification*-done`, all SRU-bugs should be listed as green in [the pending_sru page](https://people.canonical.com/~ubuntu-archive/pending-sru.html).
 
-    g. After the pending SRU page says that ubuntu-advantage-tools has been in proposed for 7 days, it is now time to ping the [current SRU vanguard](https://wiki.ubuntu.com/StableReleaseUpdates#Publishing) for acceptance of ubuntu-advantage-tools into -updates.
+    * After the pending SRU page says that ubuntu-advantage-tools has been in proposed for 7 days, it is now time to ping the SRU Reviewer for acceptance of ubuntu-advantage-tools into -updates.
```
```suggestion
    * After the pending SRU page says that `ubuntu-advantage-tools` has been in `-proposed` for 7 days, it is now time to ping the SRU Reviewer for acceptance of `ubuntu-advantage-tools` into `-updates`.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -216,20 +267,34 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
-    g. After the pending SRU page says that ubuntu-advantage-tools has been in proposed for 7 days, it is now time to ping the [current SRU vanguard](https://wiki.ubuntu.com/StableReleaseUpdates#Publishing) for acceptance of ubuntu-advantage-tools into -updates.
+    * After the pending SRU page says that ubuntu-advantage-tools has been in proposed for 7 days, it is now time to ping the SRU Reviewer for acceptance of ubuntu-advantage-tools into -updates.
 
     h. Check `rmadison ubuntu-advantage-tools` for updated version in -updates.
```
```suggestion
    h. Check `rmadison ubuntu-advantage-tools` for updated version in `-updates`.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -216,20 +267,34 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
-2. Bring in any changes that were made to the release branch into `main` via PR (e.g., changelog edits).
-3. Move any scripts added in `sru/` to a new folder in `sru/_archive` for the release.
-4. Tell CPC that there is a new version of `ubuntu-advantage-tools` in -updates for all series.
+### 5. Finalizing the release and preparing for the next one
```
```suggestion
### 5. Finalising the release and preparing for the next one
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -216,20 +267,34 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+2. Close the release PR and delete the `review-v$version` branch.
+3. Perform the steps from "Releasing to our staging PPA" above but use a `~stableppaX` suffix instead of `~rcX` in the version name, and upload to `ppa:ua-client/stable` instead of staging.
+4. Release the documentation changes
+    * Open a pull request merging the `docs-devel` branch to the `docs` branch. Ask in the ~Pro channel for a review. Some conflicts may apperar but those should be only content-related and easy enough to fix.
```
```suggestion
    * Open a pull request merging the `docs-devel` branch to the `docs` branch. Ask in the `~Pro` channel for a review. Some conflicts may appear but those should be only content-related and easy enough to fix.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -216,20 +267,34 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+    * Open a pull request merging the `docs-devel` branch to the `docs` branch. Ask in the ~Pro channel for a review. Some conflicts may apperar but those should be only content-related and easy enough to fix.
+    * Once the pull request lands, the `latest` documentation will be automatically updated.
+    * Tag the HEAD of the `docs` branch as `v<version>`. Example: `v28`.
+      * The `v` in the tag differentiates the documentation releases as they will appear in readthedocs from the software version tags in the `release` branch.
```
```suggestion
      * The `v` in the tag differentiates the documentation releases as they will appear in Read the Docs from the software version tags in the `release` branch.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -216,20 +267,34 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+    * Once the pull request lands, the `latest` documentation will be automatically updated.
+    * Tag the HEAD of the `docs` branch as `v<version>`. Example: `v28`.
+      * The `v` in the tag differentiates the documentation releases as they will appear in readthedocs from the software version tags in the `release` branch.
+    * Go to [the project in readthedocs](https://readthedocs.com/projects/canonical-ubuntu-pro-client/), to the `Versions` tab, look for the tag in the inactive versions list and click `Activate`.
```
```suggestion
    * Go to [the project in Read the Docs](https://readthedocs.com/projects/canonical-ubuntu-pro-client/), go to the `Versions` tab, look for the tag in the "inactive versions" list and click `Activate`.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -216,20 +267,34 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+      * The `v` in the tag differentiates the documentation releases as they will appear in readthedocs from the software version tags in the `release` branch.
+    * Go to [the project in readthedocs](https://readthedocs.com/projects/canonical-ubuntu-pro-client/), to the `Versions` tab, look for the tag in the inactive versions list and click `Activate`.
+      * Be sure the `Active` checkbox is checked, and the privacy level is set to `Public`.
+      * *If* there were are any changes applied only to `docs`, open a pull request merging the `docs` branch back to `docs-devel` .
```
```suggestion
      * *If* there were any changes applied only to `docs`, open a pull request merging the `docs` branch back to `docs-devel`.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -216,20 +267,34 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+5. Rebase `next-v$((version+1))` on top of `main` and open a PR to merge it. Once it is merged, delete the `next-v$((version+1))` branch.
+6. Change GitHub's "default" branch back to `main`
+    - This should automatically switch any open PRs against `next-v$((version+1))` to target `main`.
+7. Change the daily PPA build recipe to use `main`
```
```suggestion
7. Change the daily PPA build recipe to use `main`.
```

s-makin commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -216,20 +267,34 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+      * Be sure the `Active` checkbox is checked, and the privacy level is set to `Public`.
+      * *If* there were are any changes applied only to `docs`, open a pull request merging the `docs` branch back to `docs-devel` .
+5. Rebase `next-v$((version+1))` on top of `main` and open a PR to merge it. Once it is merged, delete the `next-v$((version+1))` branch.
+6. Change GitHub's "default" branch back to `main`
```
```suggestion
6. Change GitHub's "default" branch back to `main`.
```

s-makin reviewed (commented):
Just nits really. I've taken the liberty of suggesting inline code decoration for things that I suspect will trigger the spellchecker to fail once we uprate these docs to be generally visible (which I'd rather not add to the exceptions list). If you disagree with any of them, please feel free to disregard :)

orndorffgrant commented:
@renanrodrigo @lucasmoura I moved the changelog entry to after the `next-v$version` split and also added a new "hotfix" release process. LMK what you think!

renanrodrigo reviewed (approved):
Thanks much for putting all this together!
LGTM

lucasmoura commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,212 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+
+`main` is the only long-lived branch.
+
+When we reach a point in `main` where we are ready to release version 42, we do the following:
```
I think we should highlight here when we should make the changelog change, just to make it simpler to visualized the whole process

lucasmoura replied:
Should we also mention here what is the process we need for the other releases we support ? Similar to what you mentioned on the hotfix section ?

orndorffgrant replied:
Yes I'll reference that here.

orndorffgrant replied:
@lucasmoura added!

lucasmoura commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,212 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+
+### 2. Write a changelog entry
+
+Create a PR against `main` finishing the changelog entry for the new version. There should be an entry at the top for `UNRELEASED` with the version `1:1+devel`. Change the version to `$version` and change `UNRELEASED` to `$devel_release`. Also review the contents of the changelog entry and add anything that is missing. This PR should be reviewed and merged before starting the release process.
```
Why should we keep in main the `1:1+devel` entry in the changelog ? Shouldn't this entry only exist in the `next` branch ?

orndorffgrant replied:
(We discussed this in a meeting): the language here is a bit confusing. This is saying that you'll have to change the current 1:1+devel that is already there into the real release changelog entry.

`main` will have 1:1+devel during development because the daily ppa will be building from it.

```suggestion
Create a PR against `main` finishing the changelog entry for the new version. In the top entry, change `1:1+devel` to `$version` and change `UNRELEASED` to `$devel_release`. Also review the contents of the changelog entry and add anything that is missing. This PR should be reviewed and merged before starting the release process.
```

orndorffgrant commented on dev-docs/howtoguides/release_a_new_version.md:
```diff
@@ -39,174 +49,212 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
+
+
+4. Approval and upload to `devel` and `-proposed` unapproved queue
+    * Once the SRU team member gives a pre-SRU approval, create the branches for each stable release. They should be named `release-$version-$release`.
```
```suggestion
    * Once the SRU team member gives a pre-SRU approval, create backport branches for each stable release. They should be named `release-$version-$release`.
```

renanrodrigo commented:
@orndorffgrant I would say this one is good to go

orndorffgrant commented:
> @orndorffgrant I would say this one is good to go

Alright, I'll land it and we can always iterate of course.
